libbeat/monitoring/report/elasticsearch/elasticsearch.go

```diff
@@ -145,7 +145,10 @@ func makeReporter(beat beat.Info, settings report.Settings, cfg *conf.C) (report
 	}
 
 	queueConfig := conf.Namespace{}
-	conf, err := conf.NewConfigFrom("mem.events: 20")
+	conf, err := conf.NewConfigFrom(map[string]interface{}{
```

Comment:
Are we able to unit test the `makeReporter()` to ensure it can actually parse the configuration it is passed? 

Even just sticking this hard coded config into a small function and testing it would have caught this, although it would be better to test the entire function.

The tests for this are fairly lacking right now, I only see a test for `makeClientParams` https://github.com/elastic/beats/blob/d64e6321a50797aaad64d1000bfb04808317d846/libbeat/monitoring/report/elasticsearch/elasticsearch_test.go#L26

Reply:
Good idea! I added a small test to make sure the function does not return an error, let me know what you think!

Reply:
Perfect, thanks. That catches this.